### PR TITLE
Reorganise structs layout to improve mem padding

### PR DESCRIPTION
**Pull request by kuba--**
Signed-off-by: kuba-- <kuba@sourced.tech>
Closes https://github.com/src-d/gitbase/issues/436

This is a small thing, but maybe worth to consider (WDYT)?. As an example I re-organized couple structs to make memory alignment optimal - less work for GC.

Changes:
- `Server` size was 192, now 176 
- `squashReposIter` size was 80, now 72
- `squashRefIter` size was size 112, now 104
- `squashCommitFilesIter` size was 120, now 112
- `squashIndexCommitFilesIter` size was 128, now 120

### CODE REVIEW

**Comment by kuba--** [comment]
@mcarmonaa - couple readings:
- https://dave.cheney.net/2015/10/09/padding-is-hard
- https://go101.org/article/memory-layout.html
- http://www.catb.org/esr/structure-packing/
- https://syslog.ravelin.com/go-and-memory-layout-6ef30c730d51

(or just by searching about go memory alignment), but briefly - memory aligns to 8 bytes (most likely but it depends on architecture), so if you layout your structure like:
```
type MyData struct {
	a bool
	b int64
	c int32
	d bool
	e string
}
```
because of 8bytes alignment, compiler will introduce memory padding, so it will look:
```
MyData.a bool: 0-1 (size 1, align 1)
padding: 1-8 (size 7, align 0) # <---------------------
MyData.b int64: 8-16 (size 8, align 8)
MyData.c int32: 16-20 (size 4, align 4)
MyData.d bool: 20-21 (size 1, align 1)
padding: 21-24 (size 3, align 0) # <--------------------
MyData.e string: 24-40 (size 16, align 8)
```
and thats why you'll loose 10 bytes.

But if you layout your structure for instance as follows:
```
type MyData struct {
	e string
	b int64
	c int32
	a bool
	d bool
}
```
you'll loose just 2 bytes (I think the last padding is OK):
```
MyData.e string: 0-16 (size 16, align 8)
MyData.b int64: 16-24 (size 8, align 8)
MyData.c int32: 24-28 (size 4, align 4)
MyData.a bool: 28-29 (size 1, align 1)
MyData.d bool: 29-30 (size 1, align 1)
padding: 30-32 (size 2, align 0) # <----------
```

Generally, the simplest thing what I do I just layout structure from the biggest type to the smallest.

**Comment by mcarmonaa** [comment]
Thank you very much @kuba-- for such a good explanation and the links! :sparkles:

File: cmd/gitbase/command/server.go [hunk 1]
```diff
@@ -49,14 +51,11 @@ type Server struct {
...
 	ReadOnly      bool     `short:"r" long:"readonly" description:"Only allow read queries. This disables creating and deleting indexes as well." env:"GITBASE_READONLY"`
-	// SkipGitErrors disables failing when Git errors are found.
-	SkipGitErrors bool
-	// Version of the application.
-	Version string
 
-	engine *sqle.Engine
-	pool   *gitbase.RepositoryPool
-	name   string
+	SkipGitErrors bool // SkipGitErrors disables failing when Git errors are found.
```
Comment (ajnavarro) [inline cmd/gitbase/command/server.go]: This is changing the output of `gitbase server --help` command? If yes, could you change it on the documentation too please?

Comment (kuba--) [reply cmd/gitbase/command/server.go]: @ajnavarro - done

**Pull request merged**